src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,90 +1,193 @@
 
 import { makeUpdateBanner } from "api/bannerAdmin";
 
 import { AnyVenue } from "types/venues";
+import { BannerFormData } from "types/banner";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { InputField } from "components/atoms/InputField";
+
+import { ConfirmationBannerModal } from "./ConfirmationBannerModal";
+
+import "./BannerAdmin.scss";
+
+const initialBannerData = {
+  title: "",
+  content: "",
+  isActionButton: false,
+  buttonUrl: "",
+  buttonDisplayText: "",
+  isFullScreen: false,
+  isCloseButton: false,
+};
 
 interface BannerAdminProps {
   venueId?: string;
   venue: AnyVenue;
+  onClose: () => void;
 }
 
 // @debt This component is almost exactly the same as IframeAdmin, we should refactor them both to use the same generic base component
 //   BannerAdmin is the 'canonical example' to follow when we do this
-export const BannerAdmin: React.FC<BannerAdminProps> = ({ venueId, venue }) => {
-  const existingBannerMessage = venue?.bannerMessage ?? "";
-
-  const inputFieldRef = useRef<HTMLInputElement>(null);
-  const [error, setError] = useState<string | null>();
+export const BannerAdmin: React.FC<BannerAdminProps> = ({
+  venueId,
+  venue,
+  onClose,
+}) => {
+  const {
+    register,
+    handleSubmit,
+    errors,
+    reset,
+    watch,
+  } = useForm<BannerFormData>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+  const isActionButtonValue = watch("isActionButton");
+
+  const [bannerData, setBannerDate] = useState<BannerFormData>(
+    initialBannerData
+  );
+  const {
+    isShown: isShowModal,
+    show: showModal,
+    hide: hideModal,
+  } = useShowHide();
+  const [isDisabledUrlFields, setIsDisabledUrlFields] = useState(
+    venue?.banner?.isActionButton
+  );
 
-  const handleInputChange = useCallback(() => {
-    setError(null);
-  }, []);
+  useEffect(() => {
+    setIsDisabledUrlFields(isActionButtonValue);
+  }, [isActionButtonValue]);
 
   const updateBannerInFirestore = useCallback(
-    (msg?: string) => {
+    (data: BannerFormData) => {
       if (!venueId) return;
-
-      makeUpdateBanner(venueId, (errorMsg) => setError(errorMsg))(msg);
+      makeUpdateBanner(venueId)(data);
+      onClose();
     },
-    [venueId]
+    [venueId, onClose]
   );
 
-  const saveBanner = useCallback(
-    (e: React.MouseEvent<HTMLButtonElement>) => {
-      e.preventDefault();
-
-      if (!inputFieldRef.current) return;
-
-      updateBannerInFirestore(inputFieldRef.current.value);
+  const onSubmit = useCallback(
+    (data: BannerFormData) => {
+      showModal();
+      setBannerDate(data);
     },
-    [updateBannerInFirestore]
+    [showModal]
   );
 
-  const clearBanner = useCallback(() => updateBannerInFirestore(""), [
-    updateBannerInFirestore,
-  ]);
+  const clearBanner = useCallback(() => {
+    updateBannerInFirestore(initialBannerData);
```

Comment:
can we just set it to `undefined`, if we clear it?

Reply:
https://github.com/sparkletown/sparkle/pull/1440/commits/bb38651315211ee4ef4ce8b671b8d0a240cad8e8